src/ctia/schemas/core.clj

```diff
@@ -38,6 +43,144 @@
    :created java.util.Date
    (s/optional-key :modified) java.util.Date})
 
+(s/defschema RealizeFnServices
+  "Maps of service functions available for realize-fns"
+  {:ConfigService {:get-in-config (s/=>* s/Any
+                                         [[s/Any]]
+                                         [[s/Any] s/Any])}
+   :StoreService {:read-store (s/pred ifn?)} ;;varags
+   :GraphQLNamedTypeRegistryService
+   {:get-or-update-named-type-registry
+    (s/=> graphql.schema.GraphQLType
+          s/Str
+          (s/=> graphql.schema.GraphQLType))}
+   :IEncryption {:encrypt (s/=> s/Any s/Any)
+                 :decrypt (s/=> s/Any s/Any)}})
+
+(s/defn APIHandlerServices->RealizeFnServices
+  :- RealizeFnServices
+  [services :- APIHandlerServices]
+  (select-keys-in
+    services
+    [:ConfigService [:get-in-config]]
+    [:StoreService [:read-store]]
+    [:GraphQLNamedTypeRegistryService [:get-or-update-named-type-registry]]
+    [:IEncryption [:decrypt :encrypt]]))
+
+(s/defschema GraphQLRuntimeOptions
+  "A map of options to resolve a DelayedGraphQLValue"
+  {:services RealizeFnServices})
+
+(defn delayed-graphql-value?
+  "A flat predicate deciding if the argument is delayed."
+  [v]
+  (fn? v))
+
+(defn resolved-graphql-value?
+  "A flat predicate deciding if the argument is not delayed."
+  [v]
+  (not (delayed-graphql-value? v)))
+
+(s/defn MaybeDelayedRealizeFnResult
```

Comment:
I was really confused with or use of `Maybe` because in Haskell/Purescript this is about nullable types.
Here what you have is closer to an `Either`. More precisely `Either DelatedRealizeFn Result`.
So I'll let you be the judge if you could switch to Either instead of `Maybe`.

Reply:
Thanks I'll find better names.

Reply:
Simplifed to `RealizeFn` and `ResolvedRealizeFn`.